engine/connection_manager.go

```diff
@@ -71,7 +71,7 @@ func (m *connectionManager) Start() error {
 
 // Stop stops the connection manager
 func (m *connectionManager) Stop() error {
-	if m == nil {
+	if m == nil || m.conn == nil {
```

Comment:
I think `m.conn == nil` should be its own if to highlight setup process issues, similar to `IsOnline`